app/main/views/add_service.py

```
@@ -39,14 +39,30 @@ def _add_invited_user_to_service(invited_user):
     return service_id
 
 
-def _create_service(service_name, email_from):
-    service_id = service_api_client.create_service(service_name=service_name,
-                                                   message_limit=current_app.config['DEFAULT_SERVICE_LIMIT'],
-                                                   restricted=True,
-                                                   user_id=session['user_id'],
-                                                   email_from=email_from)
-    session['service_id'] = service_id
-    return service_id
+def _create_service(service_name, email_from, form):
+    try:
+        service_id = service_api_client.create_service(service_name=service_name,
+                                                       message_limit=current_app.config['DEFAULT_SERVICE_LIMIT'],
+                                                       restricted=True,
+                                                       user_id=session['user_id'],
+                                                       email_from=email_from)
+        session['service_id'] = service_id
+        return service_id, None
+    except HTTPError as e:
+        if e.status_code == 400 and e.message['name']:
+            form.name.errors.append("This service name is already in use")
+            return None, e
```

we should probably have an `else: raise` here, or we'll swallow up legit errors